Update DG with Test Cases for Manual Testing Part 2/2 (#255)
AY2223S2-CS2103T-T15-4/tp: docs/DeveloperGuide.md

### Students Page Commands
These commands should be tested on the Students Page.

#### Adding a student

1. Adding a student while the student list is empty or the student does not exist in the course list.

>    1. Test case: ` add student n/John Doe p/98765432 e/johnd@example.com sid/A1234567X`<br>

Reviewer: ```suggestion
   1. Test case: `add student n/John Doe p/98765432 e/johnd@example.com sid/A1234567X`<br>
```

      Expected: New student is added into the list. Details of the added student shown in the log box.

>    2. Other incorrect `add student` commands to try: `add`, `add student` (where the name, phone number, email or student ID is empty)<br>

Reviewer: ```suggestion
   2. Other incorrect `add student` commands to try: `add`, `add student` (where the name, phone number, email or student ID is not specified)<br>
```

      Expected: No student is added. Error details shown in the log box.

> 2. Adding a student whose student ID already exists in the list.

Reviewer 2: ```suggestion
2. Adding a student whose student ID already exists in the list.<br>
```

    1. Test case: `add student n/John p/98765432 e/johnd@example.com sid/A1234567X`, where another student with student ID A1234567X already exists in the student list<br>

>        Expected: No course is added. Error details shown in the log box.

Reviewer: ```suggestion
       Expected: No student is added. Error details shown in the log box.
```

#### Deleting a student

1. Deleting a student while all students are being shown from the Students Page.

    1. Test case: `delete student 1`<br>
       Expected: First student is deleted from the list. Details of the deleted student shown in the log box.

    2. Test case: `delete student 0`<br>
       Expected: No student is deleted. Error details shown in the log box.

   3. Other incorrect `delete student` commands to try: `delete`, `delete student x`, (where x is larger than the list size)<br>
          Expected: No student is deleted. Error details shown in the log box.

2. Deleting a student while the student list is empty on the Students Page.
    1. Test case: `delete student 1`<br>
    Expected: No student is deleted. Error details shown in the log box.

#### Copying a student's email
1. Copying a student's email to your clipboard, while there is more than one student on the Students Page.
    1. Test case: `copy 1`<br>
    Expected: First student's email is copied to your clipboard. Successful command message shown in the log box.
   2. Test case: `copy 0`<br>
   Expected: No student email is copied. Error details shown in the log box.

>    3. Other incorrect `copy` commands to try: `copy`, `copy x`, (where x is larger than the student list)

Reviewer 2: ```suggestion
   3. Other incorrect `copy` commands to try: `copy`, `copy x`, (where x is larger than the student list)<br>
```

   Expected: No student email is copied. Error details shown in the log box.
2. Copying a student's email while the student list is empty on the Students Page.
    1. Test case: `copy 1`<br>
    Expected: Nothing is copied to your clipboard. Error details shown in the log box.

#### Sorting students by their name or student ID
1. Sorting student list by their name

>     1. Test case: `sort name`

Reviewer 2: ```suggestion
    1. Test case: `sort name`<br>
```

   Expected: Student list sorted by name in alphabetical order.
2. Sorting student list by their student IDs

>     1. Test case: `sort id`

Reviewer 2: ```suggestion
    1. Test case: `sort id`<br>
```

   Expected: Student list sorted by student ID.

#### Adding or deleting a remark
1. Adding a remark to a student that does not have an existing remark.

>     1. Test case: `remark 1 Loves photography`

Reviewer 2: ```suggestion
    1. Test case: `remark 1 Loves photography`<br>
```

   Expected: Adds a remark `Loves photography` to the first student on the student list. Details of the new remark displayed in the log box.

>    2. Test case: `remark 0 Loves sci-fi movies`

Reviewer 2: ```suggestion
   2. Test case: `remark 0 Loves sci-fi movies`<br>
```

   Expected: Displays error message on the log box.

>    3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)

Reviewer 2: ```suggestion
   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student list)<br>
```

   Expected: Displays error message on the log box.
2. Adding a remark to a student that already has a remark.

>    1. Test case: `remark 1 Plays football` 

Reviewer 2: ```suggestion
   1. Test case: `remark 1 Plays football` <br>
```

      Expected: Replaces the previous remark that the first student has with `Plays football`. Details of the new remark displayed in the log box.

>    2. Test case: `remark 0 Loves sci-fi movies`

Reviewer 2: ```suggestion
   2. Test case: `remark 0 Loves sci-fi movies`<br>
```

     Expected: Displays error message on the log box.

>    3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)

Reviewer 2: ```suggestion
   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student list)<br>
```

      Expected: Displays error message on the log box.

>    

Reviewer: Maybe can add cases of deleting remark, it was mentioned in the title but not listed down

### Session Page Commands
These commands should be tested on the Session Page.

#### Adding a session
1. Adding a session while the session list is empty, or the session does not exist in the session list.

    1. Test case: `add session Tutorial1`<br>

>        Expected: New session is added into the list. Details of the added task shown in the log box.

Reviewer 2: ```suggestion
       Expected: New session is added into the list. Details of the added session shown in the log box.
```

    1. Test case: `add session Tutorial1`<br>
       Expected: No session is added. Error details shown in the log box.
   
#### Deleting a session

1. Deleting a session while all sessions are being shown from the Session Page. At least 1 session exists in the list.

    1. Test case: `delete session 1`<br>
       Expected: First session is deleted from the list. Details of the deleted session shown in the log box.

    1. Test case: `delete session 0`<br>
       Expected: No session is deleted. Error details shown in the log box.

    1. Other incorrect `delete session` commands to try: `delete`, `delete session x` (where x is larger than the list size)<br>
       Expected: Similar to previous step 1.2 under `Deleting a session`.

2. Deleting a session while the session list is empty in the Session Page.

    1. Incorrect `delete session` commands to try: `delete`, `delete session x` (where x is any number)<br>
       Expected: No session is deleted. Error details shown in the log box.

#### Editing a session

1. Editing a session which exists in the Session list.

   1. Test case: `edit session 1 Tutorial4`<br>

>       Expected: First session is edited to the new task name. Details of the edited session shown in the log box.

Reviewer 2: ```suggestion
      Expected: First session is edited to the new session name. Details of the edited session shown in the log box.
```

    1. Test case: `find session 1`<br>
       Expected: `Tutorial1` and `Lab1` is displayed. Details of the found sessions shown in the log box.

    1. Test case: `find session tutorial`<br>
       Expected: `Tutorial1` and `Tutorial2` is displayed. Details of the found sessions shown in the log box.

    1. Test case: `find session Tutorial3` (this session does not exist in the list)<br>
       Expected: The list is not filtered. Details of no found session(s) shown in the log box.

    1. Other incorrect `find session` commands to try: `find`, `find session`<br>
       Expected: The list is not filtered. Error details shown in the log box.

[Back to Instructions for Manual Testing](#appendix-instructions-for-manual-testing)

---

### Attendance Page Commands
These commands should be tested on the Attendance Page.

#### Marking selected student(s) as present

1. Marking selected student as present, with at least one student in the list.

   1. Test case: `mark 1`<br>
      Expected: First student is marked as present. Details of the student marked as present shown in the log box.

>    2. Other incorrect `mark` commands to try: `mark`, `mark x` (where x is larger than the student list size)
>       , `mark -1` or `mark 101` (where x is a valid index from the student list) <br>
>       Expected: No student is marked as present. Error details shown in the log box.

Reviewer 2: ```suggestion
   2. Other incorrect `mark` commands to try: `mark`, `mark x` (where x is larger than the student list size, `mark -1` <br>
      Expected: No student is marked as present. Error details shown in the log box.
```

Reason: `mark 101` is a correct command if index 101 exists for `at least one student`.

   
2. Marking selected students as present, with 6 students in the list.
    1. Test case: `mark 1, 4, 6`<br> 
        Expected: First, fourth and sixth students are marked as present. Details of the students marked as present shown in the log box.
    2. Test case: `mark 2, 3, 5, 7` (where there is no student in the seventh index) <br>
        Expected: No student is marked as present. Error details shown in the log box.
    3. Other incorrect `mark` commands to try: `mark`, `mark 0` <br>
       Expected: No student is marked as present. Error details shown in the log box.
   
#### Marking selected student(s) as absent
1. Marking selected student as absent, with at least one student in the list.

    1. Test case: `unmark 1`<br>
       Expected: First student is marked as absent. Details of the student marked as absent shown in the log box.

>     2. Other incorrect `unmark` commands to try: `unmark`, `unmark x` (where x is larger than the student list size)
>        , `unmark -1` or `unmark 101` (where x is a valid index from the student list) <br>
>        Expected: No student is marked as absent. Error details shown in the log box.

Reviewer 2: ```suggestion
    2. Other incorrect `unmark` commands to try: `unmark`, `unmark x` (where x is larger than the student list size), `unmark -1`<br>
       Expected: No student is marked as absent. Error details shown in the log box.
```

Reason: `unmark 101` is a correct command if index 101 exists for at least one student.
